### PR TITLE
update filter name() method

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

File: echo2_config.cc [hunk 1]
```diff
@@ -25,7 +25,7 @@ class Echo2ConfigFactory : public NamedNetworkFilterConfigFactory {
     return ProtobufTypes::MessagePtr{new Envoy::ProtobufWkt::Empty()};
   }
 
-  std::string name() override { return "echo2"; }
+  const std::string name() const override { return "echo2"; }
```
Comment (htuch) [inline echo2_config.cc]: Why is this `const std::string()`? It should either be `std::string` or `const std::string&` I think.

Comment (kyessenov) [reply echo2_config.cc]: I didn't want the filter to own its name string. Fine dropping const from std::string if you think that's better.

Comment (htuch) [reply echo2_config.cc]: Well, `const std::string` for a return value doesn't do anything so it should be dropped IMHO.

Comment (kyessenov) [reply echo2_config.cc]: Done.

**Comment by dio** [comment]
@kyessenov, sorry, does that `name()` should override https://github.com/envoyproxy/envoy/blob/11934bd081011cd7d58a9d49be4965917cce8aa0/include/envoy/server/filter_config.h#L349? Seems like it's failing?

https://circleci.com/gh/envoyproxy/envoy-filter-example/6958

```
echo2_config.cc:28:28: error: non-virtual member function marked 'override' hides virtual member function
  std::string name() const override { return "echo2"; }
                           ^
bazel-out/k8-fastbuild/bin/external/envoy/include/envoy/server/_virtual_includes/filter_config_interface/envoy/server/filter_config.h:349:23: note: hidden overloaded virtual function 'Envoy::Server::Configuration::NamedNetworkFilterConfigFactory::name' declared here: different qualifiers (unqualified vs 'const')
  virtual std::string name() PURE;
                      ^
In file included from echo2_config.cc:5:
bazel-out/k8-fastbuild/bin/external/envoy/include/envoy/registry/_virtual_includes/registry/envoy/registry/registry.h:322:5: error: field type 'Envoy::Server::Configuration::Echo2ConfigFactory' is an abstract class
  T instance_{};
    ^
echo2_config.cc:36:87: note: in instantiation of template class 'Envoy::Registry::RegisterFactory<Envoy::Server::Configuration::Echo2ConfigFactory, Envoy::Server::Configuration::NamedNetworkFilterConfigFactory>' requested here
static Registry::RegisterFactory<Echo2ConfigFactory, NamedNetworkFilterConfigFactory> registered_;
                                                                                      ^
bazel-out/k8-fastbuild/bin/external/envoy/include/envoy/server/_virtual_includes/filter_config_interface/envoy/server/filter_config.h:349:23: note: unimplemented pure virtual method 'name' in 'Echo2ConfigFactory'
  virtual std::string name() PURE;
                      ^
```

**Comment by kyessenov** [comment]
Argh, sorry. It's coupled with my PR
https://github.com/envoyproxy/envoy/pull/9391, not sure how to pull off
making both repos pass.

**Pull request merged**